src/mui/input/ReferenceArrayInput.js

```diff
@@ -172,20 +176,32 @@ export class ReferenceArrayInput extends Component {
             );
         }
 
-        if (!(referenceRecords && referenceRecords.length > 0) && !allowEmpty) {
-            return (
-                <Labeled
-                    label={
-                        typeof label === 'undefined' ? (
-                            `resources.${resource}.fields.${source}`
-                        ) : (
-                            label
-                        )
-                    }
-                    source={source}
-                    resource={resource}
-                />
-            );
+        const finalLabel =
+            typeof label === 'undefined'
+                ? `resources.${resource}.fields.${source}`
+                : label;
+        const translatedLabel = translate(finalLabel, { _: finalLabel });
+        let error =
+            matchingReferences && matchingReferences.error
+                ? translate(matchingReferences.error, {
+                      _: matchingReferences.error,
+                  })
+                : null;
+        if (
+            !error &&
+            input.value &&
+            input.value.length > 0 &&
+            !referenceRecords.length
+        ) {
+            error = translate('aor.input.references.missing');
+        }
+
+        if (matchingReferences === null) {
```

Comment:
why don't you do that *before* building the error message? It would provide an early return.